### PR TITLE
enable opt-in to case-insensitive mapping of enum values

### PR DESCRIPTION
**Pull request by earloc**
#7

### CODE REVIEW

**Comment by earloc** [comment]
Or should this be a seperate overload, in order to not break callers, when they just drop in a new Version without recompilation?

File: src/AutoMapper.Extensions.EnumMapping.Tests/EnumValueMappingByName.cs [hunk 1]
```diff
@@ -1,4 +1,7 @@
 ﻿using System;
+using System.Collections;
```
Comment (HenkKin) [inline src/AutoMapper.Extensions.EnumMapping.Tests/EnumValueMappingByName.cs:2]: removed unused usings

Comment (HenkKin) [inline src/AutoMapper.Extensions.EnumMapping.Tests/EnumValueMappingByName.cs:2]: removed unused usings

File: src/AutoMapper.Extensions.EnumMapping.Tests/EnumValueMappingByName.cs [hunk 2]
```diff
@@ -1,4 +1,7 @@
...
+using System.Collections.Generic;
```
Comment (HenkKin) [inline src/AutoMapper.Extensions.EnumMapping.Tests/EnumValueMappingByName.cs:3]: removed unused usings

File: src/AutoMapper.Extensions.EnumMapping.Tests/EnumValueMappingByName.cs [hunk 3]
```diff
@@ -1,4 +1,7 @@
...
+using System.Linq;
```
Comment (HenkKin) [inline src/AutoMapper.Extensions.EnumMapping.Tests/EnumValueMappingByName.cs:4]: removed unused usings

File: src/AutoMapper.Extensions.EnumMapping/Internal/EnumMappingFeature.cs [hunk 4]
```diff
@@ -10,6 +10,8 @@ internal class EnumMappingFeature<TSource, TDestination> : IMappingFeature, IEnu
         where TDestination : struct, Enum
     {
         protected EnumMappingType EnumMappingType = EnumMappingType.Value;
+        protected bool IgnoreCase = true;
```
Comment (HenkKin) [inline src/AutoMapper.Extensions.EnumMapping/Internal/EnumMappingFeature.cs:13]: default should be false, so remove " = true"

**Review by HenkKin — CHANGES_REQUESTED** [review]

File: src/AutoMapper.Extensions.EnumMapping/IEnumConfigurationExpression.cs [hunk 5]
```diff
@@ -15,7 +15,7 @@ public interface IEnumConfigurationExpression<in TSource, in TDestination>
         /// Map enum values by name
         /// </summary>
         /// <returns>Enum configuration options</returns>
```
[the remark this answers is not in the document]
Comment (earloc) [reply src/AutoMapper.Extensions.EnumMapping/IEnumConfigurationExpression.cs:18]: ✔️ done

File: src/AutoMapper.Extensions.EnumMapping.Tests/EnumValueMappingByName.cs (hunk 1 above)
[the remark this answers is not in the document]
Comment (earloc) [reply src/AutoMapper.Extensions.EnumMapping.Tests/EnumValueMappingByName.cs]: ✔️ done

File: src/AutoMapper.Extensions.EnumMapping.Tests/EnumValueMappingByName.cs (hunk 2 above)
[the remark this answers is not in the document]
Comment (earloc) [reply src/AutoMapper.Extensions.EnumMapping.Tests/EnumValueMappingByName.cs]: ✔️ done

File: src/AutoMapper.Extensions.EnumMapping.Tests/EnumValueMappingByName.cs (hunk 3 above)
[the remark this answers is not in the document]
Comment (earloc) [reply src/AutoMapper.Extensions.EnumMapping.Tests/EnumValueMappingByName.cs]: ✔️ done

File: src/AutoMapper.Extensions.EnumMapping.Tests/EnumValueMappingByName.cs (hunk 1 above)
Comment (earloc) [reply src/AutoMapper.Extensions.EnumMapping.Tests/EnumValueMappingByName.cs]: ✔️ done

File: src/AutoMapper.Extensions.EnumMapping.Tests/EnumValueMappingByName.cs (hunk 2 above)
Comment (earloc) [reply src/AutoMapper.Extensions.EnumMapping.Tests/EnumValueMappingByName.cs]: ✔️ done

File: src/AutoMapper.Extensions.EnumMapping.Tests/EnumValueMappingByName.cs (hunk 3 above)
Comment (earloc) [reply src/AutoMapper.Extensions.EnumMapping.Tests/EnumValueMappingByName.cs]: ✔️ done

File: src/AutoMapper.Extensions.EnumMapping/Internal/EnumMappingFeature.cs (hunk 4 above)
Comment (earloc) [reply src/AutoMapper.Extensions.EnumMapping/Internal/EnumMappingFeature.cs]: ✔️ done

**Review by HenkKin — APPROVED** [review]

**Pull request merged**